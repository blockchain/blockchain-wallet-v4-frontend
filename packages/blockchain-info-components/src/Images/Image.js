import React from 'react'
import PropTypes from 'prop-types'
import { keysIn } from 'ramda'
import styled from 'styled-components'

import Images from './Images'

export const BaseImage = styled.img`
  width: ${(props) => props.width};
  height: ${(props) => props.height};
  color: ${(props) => props.color};
`

const Image = (props) => {
  const { name, srcset, ...rest } = props
  const file = Images[name]
  const srcSet = srcset ? keysIn(srcset).map((name) => `${Images[name]} ${srcset[name]}`) : []
  if (!file) {
    return <img alt='empty-img' />
  }
  return <BaseImage src={file} srcSet={srcSet.join(', ')} {...rest} />
}

Image.defaultProps = {
  color: 'auto',
  height: 'auto',
  width: 'auto'
}

Image.propTypes = {
  color: PropTypes.string,
  height: PropTypes.string,
  name: PropTypes.string.isRequired,
  width: PropTypes.string
}

export default Image
