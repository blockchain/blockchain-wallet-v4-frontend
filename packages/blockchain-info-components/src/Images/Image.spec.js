import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Image from './Image'

describe('Image component', () => {
  it('default renders correctly', () => {
    const component = shallow(<Image name='testing' width='10px' height='10px' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
