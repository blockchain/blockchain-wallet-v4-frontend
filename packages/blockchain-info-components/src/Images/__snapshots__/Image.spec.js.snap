// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component default renders correctly 1`] = `
<img
  alt="empty-img"
/>
`;
